qiskit_experiments/framework/base_experiment.py

```diff
@@ -128,22 +128,36 @@ def run(
         circuits = transpile(self.circuits(backend), backend, **transpile_opts)
         self._postprocess_transpiled_circuits(circuits, backend, **run_options)
 
-        if isinstance(backend, LegacyBackend):
-            qobj = assemble(circuits, backend=backend, **run_opts)
-            job = backend.run(qobj)
+        # Run experiment jobs
+        max_experiments = getattr(backend.configuration(), "max_experiments", None)
+        if max_experiments and len(circuits) > max_experiments:
```

Comment:
This should be taken also from run options. If we use pulse gate and scan a parameter of very long pulse, this job will quickly consume the waveform memory on hardware and raises the memory overflow error on a backend. To prevent this, we often split a job to keep total waveform number sufficiently small. For example, CR Hamiltonian tomography often runs into this situation.